File: backend/src/schemas/category.ts
import z from 'zod';

export const CategoryRes = z.object({
  id: z.bigint(),
  name: z.string(),
  img: z.string().nullable().optional(),
  created: z.date(),
  updated: z.date(),
  created_by: z.bigint().nullable(),
  updated_by: z.bigint().nullable(),
  deleted: z.boolean(),
  active: z.boolean(),
});
export type CategoryResType = z.TypeOf<typeof CategoryRes>;

File: backend/app/schemas/category.py
from pydantic import BaseModel
from typing import Optional

class CategoryCreate(BaseModel):
    name: str

class CategoryUpdate(BaseModel):
    name: Optional[str] = None
    img: Optional[str] = None

class CategoryOut(BaseModel):
    id: str
    name: str
    img: Optional[str] = None
    deleted: bool
    active: bool

    model_config = {"from_attributes": True}
